### ISSUE TITLE
awx-uwsgi stderr

### ISSUE THREAD

**Issue by vailenwallaces** [ansible/awx#14192]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

After upgrading docker-compose "installation" from 21.14.1 to any other 22.X.X (22.4.0 was the last version that i try) i got the message : 

awx-uwsgi stderr | [pid: 20252|app: 0|req: 683/38353] XXXXXXX () {34 vars in 434 bytes} [Sun Jul  2 00:05:51 2023] GET /api/v2/ping/?active_node_only=true => generated 472 bytes in 46 msecs (HTTP/1.1 200) 14 headers in 465 bytes (1 switches on core 0)
awx-uwsgi stderr |

I think this is a normal message and not an error message , the message is generated by our LoadBalancer. 

Furthermore, after the upgrade we get these messages once, I think it is also only information, not error messages. But also here I am not sure.

awx-dispatcher stderr | 2023-06-29 18:50:22,871 WARNING  [-] awx.main.dispatch.periodic periodic beat started
awx-dispatcher stderr |
awx-dispatcher stderr | 2023-06-29 18:50:22,898 INFO     [-] awx.main.dispatch Running worker dispatcher listening to queues ['tower_broadcast_all', 'tower_settings_change', 'awx_1']
awx-dispatcher stderr |
awx-receiver stderr | 2023-06-29 18:50:22,912 INFO     [-] awx.main.commands.run_callback_receiver Callback receiver started with pid=216
awx-receiver stderr |
awx-wsrelay stderr | 2023-06-29 18:50:23,037 INFO     [-] awx.main.wsrelay Active instance with hostname awx_1 is registered.
awx-wsrelay stderr |
2023-06-29 18:50:23,225 INFO waiting for awx-rsyslogd to stop
2023-06-29 18:50:23,227 INFO stopped: awx-rsyslogd (exit status 0)
2023-06-29 18:50:23,230 INFO spawned: 'awx-rsyslogd' with pid 405
awx-daphne stderr | 2023-06-29 18:50:23,246 INFO     [-] daphne.cli Starting server at tcp:port=8051:interface=127.0.0.1
awx-daphne stderr | 2023-06-29 18:50:23,246 INFO     Starting server at tcp:port=8051:interface=127.0.0.1
awx-daphne stderr |
awx-daphne stderr | 2023-06-29 18:50:23,248 INFO     [-] daphne.server HTTP/2 support not enabled (install the http2 and tls Twisted extras)
awx-daphne stderr | 2023-06-29 18:50:23,248 INFO     HTTP/2 support not enabled (install the http2 and tls Twisted extras)
awx-daphne stderr | 2023-06-29 18:50:23,248 INFO     [-] daphne.server Configuring endpoint tcp:port=8051:interface=127.0.0.1
awx-daphne stderr | 2023-06-29 18:50:23,248 INFO     Configuring endpoint tcp:port=8051:interface=127.0.0.1
awx-daphne stderr |
awx-daphne stderr | 2023-06-29 18:50:23,249 INFO     [-] daphne.server Listening on TCP address 127.0.0.1:8051
awx-daphne stderr | 2023-06-29 18:50:23,249 INFO     Listening on TCP address 127.0.0.1:8051
awx-daphne stderr |
awx-uwsgi stderr | [uWSGI] getting INI configuration from /etc/tower/uwsgi.ini
awx-uwsgi stderr |
awx-uwsgi stderr | *** Starting uWSGI 2.0.21 (64bit) on [Thu Jun 29 18:50:23 2023] ***
awx-uwsgi stderr | compiled with version: 11.4.1 20230605 (Red Hat 11.4.1-2) on 28 June 2023 15:54:37
awx-uwsgi stderr | os: Linux-5.4.0-153-generic #170-Ubuntu SMP Fri Jun 16 13:43:31 UTC 2023
awx-uwsgi stderr | nodename: awx_1
awx-uwsgi stderr | machine: x86_64
awx-uwsgi stderr | clock source: unix
awx-uwsgi stderr | detected number of CPU cores: 4
awx-uwsgi stderr | current working directory: /awx_devel
awx-uwsgi stderr | detected binary path: /var/lib/awx/venv/awx/bin/uwsgi
awx-uwsgi stderr | !!! no internal routing support, rebuild with pcre support !!!
awx-uwsgi stderr |
awx-uwsgi stderr | your memory page size is 4096 bytes
awx-uwsgi stderr | detected max file descriptor number: 1048576
awx-uwsgi stderr | lock engine: pthread robust mutexes
awx-uwsgi stderr | thunder lock: disabled (you can enable it with --thunder-lock)
awx-uwsgi stderr | uwsgi socket 0 bound to TCP address 127.0.0.1:8050 fd 3
awx-uwsgi stderr | Python version: 3.9.16 (main, Mar  7 2023, 00:00:00)  [GCC 11.3.1 20221121 (Red Hat 11.3.1-4)]
awx-uwsgi stderr |
awx-uwsgi stderr | *** Python threads support is disabled. You can enable it with --enable-threads ***
awx-uwsgi stderr | Python main interpreter initialized at 0x14494e0
awx-uwsgi stderr | your server socket listen backlog is limited to 100 connections
awx-uwsgi stderr | your mercy for graceful operations on workers is 60 seconds
awx-uwsgi stderr |
awx-uwsgi stderr | *** Python threads support is disabled. You can enable it with --enable-threads ***
awx-uwsgi stderr | Python main interpreter initialized at 0x14494e0
awx-uwsgi stderr | your server socket listen backlog is limited to 100 connections
awx-uwsgi stderr | your mercy for graceful operations on workers is 60 seconds
awx-uwsgi stderr |
awx-uwsgi stderr | mapped 609552 bytes (595 KB) for 5 cores
awx-uwsgi stderr | *** Operational MODE: preforking ***
awx-uwsgi stderr |
awx-uwsgi stderr | *** uWSGI is running in multiple interpreter mode ***
awx-uwsgi stderr | spawned uWSGI master process (pid: 407)
awx-uwsgi stderr | spawned uWSGI worker 1 (pid: 408, cores: 1)
awx-uwsgi stderr | spawned uWSGI worker 2 (pid: 409, cores: 1)
awx-uwsgi stderr | spawned uWSGI worker 3 (pid: 410, cores: 1)
awx-uwsgi stderr | spawned uWSGI worker 4 (pid: 411, cores: 1)
awx-uwsgi stderr | spawned uWSGI worker 5 (pid: 412, cores: 1)
awx-uwsgi stderr |
awx-uwsgi stderr | mounting awx.wsgi:application on /
awx-uwsgi stderr |
awx-uwsgi stderr | mounting awx.wsgi:application on /
awx-uwsgi stderr |
awx-uwsgi stderr | mounting awx.wsgi:application on /
awx-uwsgi stderr |
awx-uwsgi stderr | mounting awx.wsgi:application on /
awx-uwsgi stderr |
awx-uwsgi stderr | mounting awx.wsgi:application on /
awx-uwsgi stderr |
2023-06-29 18:50:24,978 INFO success: awx-rsyslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
awx-uwsgi stderr | WSGI app 0 (mountpoint='/') ready in 2 seconds on interpreter 0x14494e0 pid: 410 (default app)
awx-uwsgi stderr |
awx-uwsgi stderr | WSGI app 0 (mountpoint='/') ready in 2 seconds on interpreter 0x14494e0 pid: 409 (default app)
awx-uwsgi stderr |
awx-uwsgi stderr | WSGI app 0 (mountpoint='/') ready in 2 seconds on interpreter 0x14494e0 pid: 408 (default app)
awx-uwsgi stderr |
awx-uwsgi stderr | WSGI app 0 (mountpoint='/') ready in 2 seconds on interpreter 0x14494e0 pid: 412 (default app)
awx-uwsgi stderr |
awx-uwsgi stderr | WSGI app 0 (mountpoint='/') ready in 2 seconds on interpreter 0x14494e0 pid: 411 (default app)
awx-uwsgi stderr |
awx-uwsgi stderr | [pid: 409|app: 0|req: 1/1] XXXXXXX () {34 vars in 434 bytes} [Thu Jun 29 18:50:34 2023] GET /api/v2/ping/?active_node_only=true => generated 472 bytes in 296 msecs (HTTP/1.1 200) 14 headers in 465 bytes (1 switches on core 0)
awx-uwsgi stderr |

### AWX version

22.4.0

### Select the relevant components

- [X] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

docker development environment

### Modifications

no

### Web browser

Edge

### Steps to reproduce

Upgrade from 21.14.1 to any 22.X.X Version

### Expected results

No Error messages on docker logs tools_awx_1

### Actual results

awx-uwsgi stderr | [pid: 20252|app: 0|req: 683/38353] XXXXXXX () {34 vars in 434 bytes} [Sun Jul  2 00:05:51 2023] GET /api/v2/ping/?active_node_only=true => generated 472 bytes in 46 msecs (HTTP/1.1 200) 14 headers in 465 bytes (1 switches on core 0)
awx-uwsgi stderr |

**Comment by stefankaufmann** [comment]
same for me. 

I am setting up awx right now and stumbled across those log lines as well. Should probably be logged to stdout rather than stderr

**Comment by djyasin** [comment]
@vailenwallaces thank you for being a valued member of our community! We would like to gather a bit more information from you. Are you encountering any other error messages aside from this one that you're seeing in stderr? Any supporting information you can provide would be very helpful. Thank you for your time!

**Comment by vailenwallaces** [comment]
@djyasin  I don't see any other problems. I have also written all the error messages I found in the ticket

**Comment by relrod** [comment]
@vailenwallaces

> I think this is a normal message and not an error message , the message is generated by our LoadBalancer.
>
> Furthermore, after the upgrade we get these messages once, I think it is also only information, not error messages. But also here I am not sure.

You are correct that these are normal messages and nothing to be concerned about. You are running the development environment, so it is reasonable to expect some amount of verbose output.

Some processes, such as uwsgi, default to logging to stderr. This is standard practice, as most of these messages are diagnostic in nature and don't really fit into the purpose of stdout. So I think this is intentional.

In the development environment, in practice, it makes little difference as everything gets redirected to stdout anyway, at least until very recent versions of `docker-compose` which I think makes an attempt to separate the two streams.

Are there practical issues you are running into as a result of this?

**Comment by vailenwallaces** [comment]
Well the practical reason is that in some environments I don't have a kubernetes cluster and this is the "easiest" way to use an AWX in an AirGab environment.  By simply transporting the built images.  To my great regret, the simple "docker" container installations have been canceled.  And for an own minicube or k3s there is a lack of know how in my environment.  So yes I am aware that I am using a development build here, but I would have liked to distinguish the "actual" errors from simple "outputs" as it was possible before version 22.X.X :-)

**Comment by relrod** [comment]
@vailenwallaces The development environment is... just that, though. We intentionally don't support using it for purposes other than developing AWX itself.

Also, the two environments behave the same way in this respect (and `kubectl logs` will munge stderr and stdout together too).

And the messages you are seeing intentionally go to stderr, because they are diagnostic messages, not output for users to see. This is pretty conventional - there is unfortunately no stdinfo stream for "not error, but also not user-facing output" - so programs often use stderr for diagnostic messages.

So I don't think there's a plan to change this.

But I'm curious about:

>  as it was possible before version 22.X.X

What is the old behavior you were seeing? I don't think we intentionally made any change to this between 21 and 22. I would expect that in 21 it was stderr, too.

**Comment by vailenwallaces** [comment]
Yes I know that docker is "only" the dev environment and that is also a risk I have to take. 
In version 21.14... it was stdout and not stderr. Some error message of ui in previous versions were issued on stderr, so I could identify them better.
